sdk/test/trace/tracer_test.cc

```diff
@@ -132,6 +152,30 @@ TEST(Tracer, StartSpanSampleOff)
   ASSERT_EQ(0, span_data->GetSpans().size());
 }
 
+TEST(Tracer, StartSpanCustomIdGenerator)
+{
+  IdGenerator *id_generator = new MockIdGenerator();
+  std::unique_ptr<InMemorySpanExporter> exporter(new InMemorySpanExporter());
+  std::shared_ptr<InMemorySpanData> span_data = exporter->GetData();
+  auto tracer = initTracer(std::move(exporter), new AlwaysOnSampler(), id_generator);
+
+  tracer->StartSpan("span 1")->End();
+  auto spans          = span_data->GetSpans();
+  auto &cur_span_data = spans.at(0);
+
+  char rec_trace[32], exp_trace[32];
+  char rec_span[16], exp_span[16];
+
+  cur_span_data->GetTraceId().ToLowerBase16(rec_trace);
+  id_generator->GenerateTraceId().ToLowerBase16(exp_trace);
```

Comment:
I think we can compare ids directly? Like `EXPECT_EQ(cur_span_data->GetTraceId(), id_generator->GenerateTraceId());`?

Reply:
Good point. It is bitwise comparison of `TraceId` data members ( array of uint8_t ) and it should work.